forge-game/src/main/java/forge/game/trigger/TriggerChangesZoneAll.java

```diff
@@ -31,6 +33,19 @@ public boolean performTest(Map<AbilityKey, Object> runParams) {
             return false;
         }
 
+        if (hasParam("FirstTime")) {
+            CardCollectionView lastState = CardCollection.EMPTY;
+            if ("Battlefield".equals(getParam("Destination"))) {
+                lastState = table.getLastStateBattlefield();
+            } else if ("Graveyard".equals(getParam("Destination"))) {
+                lastState = table.getLastStateGraveyard();
+            }
+            // currently only for Crawling Sensation
+            if (CardLists.getValidCardCount(lastState, getParam("ValidCards") + "+ThisTurnEntered", getHostCard().getController(), getHostCard(), this) > 0) {
+                return false;
+            }
+        }
+
```

Comment:
i don't think LastStateGraveyard words there as you think

for example, what if i remove all lands that was sent there
then play Crawling Sensation
and then mill new lands?

because LastStateGraveyard is empty, this condition is successful

Reply:
Instead, we probably need to check for:

```
list = getTurnEntered("Graveyard", "Any")
list.removeAll(table.cards)
list.isEmpty()
```

Reply:
yea good idea
seems to work